### PR TITLE
Add handling sklearn version in `OneHotEncoderTransform` and `HierarchicalClustering`

### LINKED ISSUE #497: [BUG] OneHotEncoderTransform got an unexpected keyword argument 'sparse'
### 🐛 Bug Report

It is told that `etna` would work with sklearn version up to 2.0.0

However, I got scikit-learn==1.5.2
OneHotEncoder in sklearn has argument `sparse_output` instead of `sparse` (since version 1.2)

Either dependency version should be restricted, or code should be updated

### Expected behavior

OneHotEncoderTransform should return transformed data matrix

### How To Reproduce

pip install sklearn==1.5.2 
pip install etna==2.9.0
```
ohe = OneHotEncoderTransform(
    in_column="categorical_columns",
    out_column="OneHot",
    return_type="numeric"
)
```

get
```
    [216](.../etna/transforms/encoders/categorical.py:216) self.out_column = out_column
    [217](.../etna/transforms/encoders/categorical.py:217) self.return_type = ReturnType(return_type)
--> [218](.../etna/transforms/encoders/categorical.py:218) self.ohe = preprocessing.OneHotEncoder(handle_unknown="ignore", sparse=False, dtype=int)
    [219](.../etna/transforms/encoders/categorical.py:219) self.in_column_regressor: Optional[bool] = None

TypeError: OneHotEncoder.__init__() got an unexpected keyword argument 'sparse'
```

### Environment

python==3.10.11
OS == windows 10

### Additional context

_No response_

### Checklist

- [X] Bug appears at the latest library version

### PR DESCRIPTION
**Pull request by d-a-bunin**
## Before submitting (must do checklist)

- [ ] Did you read the [contribution guide](https://github.com/etna-team/etna/blob/master/CONTRIBUTING.md)?
- [ ] Did you update the docs? We use Numpy format for all the methods and classes.
- [ ] Did you write any new necessary tests?
- [ ] Did you update the [CHANGELOG](https://github.com/etna-team/etna/blob/master/CHANGELOG.md)?

## Proposed Changes
Add workaroud for working in sklearn >= 1.4.

## Closing issues
Closes #497 and #516

### CODE REVIEW

File: poetry.lock [hunk 1]
```diff
@@ -4740,6 +4740,41 @@ docs = ["Pillow (>=7.1.2)", "matplotlib (>=2.2.3)", "memory-profiler (>=0.57.0)"
 examples = ["matplotlib (>=2.2.3)", "pandas (>=0.25.0)", "scikit-image (>=0.14.5)", "seaborn (>=0.9.0)"]
 tests = ["black (>=21.6b0)", "flake8 (>=3.8.2)", "matplotlib (>=2.2.3)", "mypy (>=0.770)", "pandas (>=0.25.0)", "pyamg (>=4.0.0)", "pytest (>=5.0.1)", "pytest-cov (>=2.9.0)", "scikit-image (>=0.14.5)"]
 
+[[package]]
```
Comment (egoriyaa) [inline poetry.lock:4743]: I can't understand why poetry is doing this with `scipy` and `statsmodels`...

**Comment by egoriyaa** [comment]
Changelog?

File: poetry.lock (hunk 1 above)
Comment (d-a-bunin) [reply poetry.lock:4743]: This happenned when I ran `poetry update scikit-learn`.

Comment (d-a-bunin) [reply poetry.lock:4743]: The strange thing is that is added two versions of `scipy` here. But in a current `master` we have two versions of `statsmodels`.

Comment (d-a-bunin) [reply poetry.lock:4743]: I have a suspicion that it uses two different versions for different python versions. I'll probably revert update in lock in this PR.

Comment (egoriyaa) [reply poetry.lock:4743]: Why do you run `poetry update scikit-learn`?

**Review by egoriyaa — APPROVED** [review]

**Pull request merged**